compiler/src/middle_end/linearize.re

```diff
@@ -182,10 +182,20 @@ let transl_const =
         [BLet(tmp, Comp.number(Const_number_bigint(data)), Nonglobal)]
       ),
     )
+  | Const_int8(i) =>
+    Right(with_bind("int8", tmp => [BLet(tmp, Comp.int8(i), Nonglobal)]))
+  | Const_int16(i) =>
+    Right(with_bind("int16", tmp => [BLet(tmp, Comp.int16(i), Nonglobal)]))
   | Const_int32(i) =>
     Right(with_bind("int32", tmp => [BLet(tmp, Comp.int32(i), Nonglobal)]))
   | Const_int64(i) =>
     Right(with_bind("int64", tmp => [BLet(tmp, Comp.int64(i), Nonglobal)]))
+  | Const_uint8(i) =>
+    Right(with_bind("uint8", tmp => [BLet(tmp, Comp.uint8(i), Nonglobal)]))
+  | Const_uint16(i) =>
+    Right(
+      with_bind("uint16", tmp => [BLet(tmp, Comp.uint16(i), Nonglobal)]),
+    )
```

Comment:
These shouldn't need bindings, with the `Left` case being fine. It's the same for chars, but it looks like that was never updated. No need the special Comp helpers for the small ints!

Reply:
Removing the binding for chars causes compilation to fail [here](https://github.com/grain-lang/grain/blob/64e80b69b36a058e4883f4e22b3b9dc0735911ce/compiler/src/codegen/transl_anf.re#L528). Should I add `MConstChar` to the mashtree?

Reply:
Yup, we should have done that when we moved Char from the heap to the stack, but we missed it.

Reply:
One of the downstream effects of removing bindings for small ints (combined with the fact that the functions in the stdlib modules are normal arithmetic operators e.g. `+`, `-`, etc.) is that they now go through constant folding in `optimize_simple_expressions.re`. This essentially means that these standard library function would never actually get called in the stdlib tests, making them basically useless.

Furthermore, I believe the fact that this happens is technically a bug. For example at the moment if I write this code:
```
let (+) = (a: Number, b: Number) => toString(a) ++ " plus " ++ toString(b)
let x = 1 + 2
print(x)
```
The output would be `3` because of the constant folding even though it should be `"1 plus 2"`